### PR TITLE
feat(trajectory_follower): add min_prediction_length to mpc

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

- When stopped and given a curve trajectory, the steer command value of mpc can not be large enough. 
  - This is because precition_length is obtained from the distance traveled in a certain time with the speeds. 
- Therefore,  `min_prediction_length` was added to look some distance ahead even when stopped.

example where the steer command is not large enough and the vehicle cannot follow the trajectory
https://user-images.githubusercontent.com/39142679/177076242-8d7b07a3-c3f7-458f-a642-652759e6ee9a.mp4

## Tests performed

checked that the vehicle can follow curve trajectory(like video above) with psim and real vehicle.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: control/trajectory_follower/include/trajectory_follower/mpc.hpp [hunk 1]
```diff
@@ -222,21 +226,24 @@ class TRAJECTORY_FOLLOWER_PUBLIC MPC
...
   /**
    * @brief generate MPC matrix with trajectory and vehicle model
    * @param [in] mpc_matrix parameters matrix to use for optimization
    * @param [in] x0 initial state vector
    * @param [out] Uex optimized input vector
    */
   bool8_t executeOptimization(
-    const MPCMatrix & mpc_matrix, const Eigen::VectorXd & x0, Eigen::VectorXd * Uex);
+    const MPCMatrix & mpc_matrix, const Eigen::VectorXd & x0, Eigen::VectorXd * Uex,
+    const float64_t max_dt);
```
Comment (rej55) [inline control/trajectory_follower/include/trajectory_follower/mpc.hpp:239]: For clarity, output arguments should place after the input arguments.

File: control/trajectory_follower/include/trajectory_follower/mpc.hpp [hunk 2]
```diff
@@ -222,21 +226,24 @@ class TRAJECTORY_FOLLOWER_PUBLIC MPC
...
   /**
    * @brief resample trajectory with mpc resampling time
    */
   bool8_t resampleMPCTrajectoryByTime(
     float64_t start_time, const trajectory_follower::MPCTrajectory & input,
-    trajectory_follower::MPCTrajectory * output) const;
+    trajectory_follower::MPCTrajectory * output, float64_t & max_dt,
+    const geometry_msgs::msg::Pose & current_pose) const;
```
Comment (rej55) [inline control/trajectory_follower/include/trajectory_follower/mpc.hpp:246]: Same here.
> For clarity, output arguments should place after the input arguments.

**Review by rej55 — COMMENTED** [review]
I have some minor comments.
Could you check it?

File: control/trajectory_follower/include/trajectory_follower/mpc.hpp (hunk 2 above)
Comment (kosuke55) [reply control/trajectory_follower/include/trajectory_follower/mpc.hpp]: fix in https://github.com/autowarefoundation/autoware.universe/pull/1171/commits/d1eea82ba0fcfac85ac9453da9b4097acf9eaffc

**Comment by kosuke55** [comment]
@rej55 
yes, It can run. Could your share error message?
How about adding `min_prediction_length` add your config or rebuild..?
(sorry for no launch PR, I will send it)

**Comment by rej55** [comment]
It runs with `mpc_min_prediction_length`.
Thank you for your comment!

**Review by rej55 — APPROVED** [review]
LGTM

**Pull request merged**